### PR TITLE
Feature/2850 handle TLM at Header.jsx

### PR DESCRIPTION
**Pull request by gbonaspetti**
#2850 

## Checkpoints

The goal of this section is to help code reviewers and the merge request author to do the required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

File: frontend/src/container/AdminAccount.jsx [hunk 1]
```diff
@@ -120,11 +120,14 @@ export class Account extends React.Component {
...
-          ? { ...ws, memberList: ws.memberList.map(member => (
+          ? {
+            ...ws,
+            memberList: ws.memberList.map(member => (
               member.id === Number(state.userToEditId)
                 ? { ...member, doNotify: data.member.do_notify }
                 : member
-          ))}
+            ))
+          }
```
Comment (raphj) [inline frontend/src/container/AdminAccount.jsx:130]: Ah ah, I hesitated to suggest you this indentation and told myself "no, this is too strict".
This looks good.
:+1: :+1:

File: frontend/test/src/container/Header.spec.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    },
+    lang: [],
+    system: {
+      config: {
+        email_notification_activated: true
+      }
+    },
+    location: {
+      pathname: {
+        startsWith: () => {}
```
Comment (Skylsmoi) [inline frontend/test/src/container/Header.spec.js:53]: startsWith() is a function of the String prototype, you shouldn't have to mock it as long as it is used on a string

Comment (gbonaspetti) [reply frontend/test/src/container/Header.spec.js]: Changed

**Comment by metzgegu** [comment]
In the current implementation, when creating a new user and connect with it for the first time (only on the first connection), this errors appears : 

image

File: frontend/src/container/Header.jsx [hunk 3]
```diff
@@ -33,12 +34,31 @@ import { Link } from 'react-router-dom'
...
+      { entityType: TLM_ET.USER, coreEntityType: TLM_CET.MODIFIED, handler: this.handleUserModified }
+    ])
+  }
+
+  handleUserModified = data => {
+    const { props } = this
+    if (props.user.userId !== data.user.user_id) return
+
+    if (props.user.lang !== data.user.lang) props.dispatch(setUserLang(data.user.lang))
+    else props.dispatch(updateUser(data.user))
```
Comment (Skylsmoi) [inline frontend/src/container/Header.jsx:59]: with the PR https://github.com/tracim/tracim/pull/3194, this should be handled by the new component ReduxTlmDispatcher

You will have to update this PR once https://github.com/tracim/tracim/pull/3194 is merged

Comment (Skylsmoi) [reply frontend/src/container/Header.jsx:59]: update: the pr is merged

**Comment by gbonaspetti** [comment]
To quality team:
- Test the change of language (with a logged and a not-logged user).
- Test user changes (especially changing his public name to see if it changes correctly in the header).

**Pull request merged**